phpunit.xml.dist

```diff
@@ -22,24 +22,37 @@
     <php>
         <ini name="error_reporting" value="-1" />
 
-        <!-- Test connection parameters -->
-        <!-- Uncomment, otherwise SQLite runs
+        <!-- "Real" test database -->
+        <var name="db_driver" value="pdo_sqlite"/>
+        <var name="db_memory" value="true"/>
+        <!-- to use another database driver / credentials, provide them like so:
         <var name="db_driver" value="pdo_mysql"/>
         <var name="db_host" value="localhost" />
-        <var name="db_port" value="3306"/>
         <var name="db_user" value="root" />
         <var name="db_password" value="" />
         <var name="db_dbname" value="doctrine_tests" />
-        -->
+        <var name="db_port" value="3306"/>-->
         <!--<var name="db_event_subscribers" value="Doctrine\DBAL\Event\Listeners\OracleSessionInit">-->
 
-        <!-- Privileged user connection parameters. Used to create and drop the test database -->
-        <var name="tmpdb_driver" value="pdo_mysql"/>
-        <var name="tmpdb_host" value="localhost" />
-        <var name="tmpdb_port" value="3306"/>
-        <var name="tmpdb_user" value="root" />
-        <var name="tmpdb_password" value="" />
-        <var name="tmpdb_dbname" value="doctrine_tests_tmp" />
+        <!--
+         At the start of each test run, we will drop and recreate the test database.
```

Comment:
This comment and "if you prefer" seem to contradict each other. The test suite needs a privileged connection to create the test database. So if a user prefers to not provide the privileged connection configuration, the test suite won't work.

This is why all platforms have both configurations.

Reply:
Apologies but that is not correct. As discussed in doctrine/orm#8532 neither dbal nor orm actually *use* a separate `privileged_db` connection in any of the out-of-the-box configurations (except Oracle in dbal).

If `privileged_db` values are not specified, the TestUtil class falls back to using the `db_xx` values, which are assumed to have root-like access. Note that this has not changed in this pull, only the naming of the `tmpdb_` values and the documentation to clarify that these are entirely optional and only required if the user wants/needs to provide a different set of credentials to create/drop the DB than the ones that will be used for the rest of the tests.

Reply:
> neither dbal nor orm actually use a separate privileged_db connection in any of the out-of-the-box configurations (except Oracle in dbal).

Are you talking about the logic introduced in https://github.com/doctrine/dbal/pull/4067? 
> All so-called temporary connections look identical to the primary ones with the exception that the temporary one doesn't usually have a database name specified (creating a database is the primary purpose of the temporary connection). Instead of requiring the parameters to be specified explicitly, in the absence of the parameters, the test suite can fall back to the primary connection parameters **and ignore the database**.

So these are two different connections built from the same set of parameters. But it's not possible for every driver, so it's not a matter of the user preference. The proposed wording is misleading.

Reply:
No offence intended, I have no idea of the history of the logic - though from a skim of the PR you mention, yes.

The wording I've proposed is my best effort to understand the meaning and function of parameters that were IMO confusingly named and not documented anywhere at all. It appears from doctrine/orm#8532 that there are core team members that were also unclear on why/how these work, and that this was probably the root cause of ORM not running CI against any real databases for months through a period of massive automated changes to the codebase.

I can see that the Oracle driver uses two different user accounts (with every other aspect of the `privileged_` and `db_` params identical). I have no idea why it can't just use `system` for both connections the same way every other driver just uses `root` or equivalent.

Obviously if you can explain it better I'd welcome alternative wording, but I'm not sure I'll be able to figure that out on my own.